src/WebUI/Workloads/OtherWorkloadsTable.tsx

```diff
@@ -391,21 +393,33 @@ export class OtherWorkloads extends React.Component<IOtherWorkloadsProperties, I
     }
 
     private _onImageClick = (imageId: string, itemUid: string = ""): void => {
-        const imageService = KubeFactory.getImageService();
-        imageService && imageService.getImageDetails(imageId).then(imageDetails => {
-            if (imageDetails) {
-                const payload: ISelectionPayload = {
+        const showImageDetails = (imageDetails: IImageDetails): void => {
+            const payload: ISelectionPayload = {
                     item: imageDetails,
                     itemUID: itemUid,
                     showSelectedItem: true,
                     selectedItemType: SelectedItemKeys.ImageDetailsKey
                 };
                 this._selectionActionCreator.selectItem(payload);
-            }
-        });
+        }
+
+        let imageDetails: IImageDetails | undefined = this._imageDetailsStore.getImageDetails(imageId);
```

Comment:
can be kept in one place.

 let imageDetails: IImageDetails | undefined = this._imageDetailsStore.getImageDetails(imageId);
        if (imageDetails) {
            showImageDetails(imageDetails);
        }
        else {
            const imageService = KubeFactory.getImageService();
            imageService && imageService.getImageDetails(imageId).then(imageDetails => {
                if (imageDetails) {
                    ActionsCreatorManager.GetActionCreator<ImageDetailsActionsCreator>(ImageDetailsActionsCreator).setImageDetails(imageDetails);
                    showImageDetails(imageDetails);
                }
            });
        }